qiskit/pulse/builder.py

```diff
@@ -1730,6 +1730,64 @@ def barrier(*channels_or_qubits: Union[chans.Channel, int]):
 
 
 # Macros
+def macro(func: Callable):
```

Comment:
how does this sit with `pulse.macros`?

Reply:
`pulse.macros` are the implementation of macro functionality at the schedule level. I think it is fine to have both, and reuse the schedule-level macros from within the corresponding build macro equivalents. Ideally, if the two interfaces were separated within a `pulse.ir` and `pulse.builder` packages, it would be clearer that they are each two parts of the same coin for different representations.